src/recipe/parser/requirements.rs

```diff
@@ -65,6 +65,11 @@ impl Requirements {
         self.run_constrained.as_slice()
     }
 
+    /// Get all requirements at build time in one iterator.
+    pub fn all_build_time(&self) -> impl Iterator<Item = &Dependency> {
+        self.build.iter().chain(self.host.iter())
```

Comment:
There is an extra requirement that would be needed for all packages that are `pin_subpackage(...)` - those can appear in the run_exports as well as any other requirement sections.